### PR TITLE
WIP: Improve interop with cats effect documentation

### PR DESCRIPTION
**Pull request by khajavi**
I've updated this documentation to be more straightforward for begginers.
- [x] ZStream
- [x] ZQueue
- [x] ZManaged
- [x] STM
- [x] Doobie
- [x] http4s

### CODE REVIEW

**Comment by khajavi** [comment]
@jdegoes 
It's ready to be review.

File: docs/howto/interop/with-cats-effect.md [hunk 1]
```diff
@@ -13,60 +13,375 @@ libraryDependencies += "dev.zio" %% "zio-interop-cats" % "<version>"
...
 
-```scala
+```scala mdoc:silent
 import zio.interop.catz._
 ```
 
 ## Cats Effect Instances
 
-ZIO integrates with Typelevel libraries by providing instances of Cats Effect type classes. These type classes are used by `fs2`, `doobie` and `http4s`, and a variety of other libraries in the functional Scala ecosystem.
+ZIO integrates with Type-level libraries by providing instances of Cats Effect type classes. These type classes are used by `fs2`, `doobie`, `http4s`, and a variety of other libraries in the functional Scala ecosystem.
```
Comment (jdegoes) [inline docs/howto/interop/with-cats-effect.md:22]: "Typelevel" is one word.

**Comment by khajavi** [comment]
Some ongoing works:
- [ ] Migrate examples to CE3
- [ ] Add fs2 streaming example as a separate one.
- [ ] Cover both sides of conversion in each section (from ZIO to Cats Effect or vice versa)

**Comment by adamgfraser** [comment]
@khajavi It may be good to have separate sections on migrating from Cats Effect 2 and Cats Effect 3.

**Comment by khajavi** [comment]
> @khajavi It may be good to have separate sections on migrating from Cats Effect 2 and Cats Effect 3.

Good point! But how to manage maintainable running code with mdoc?
In the `build.sbt`, we can only specify one version, Cats Effect 2 or Cats Effect 3. Do you have any idea how to manage that?
Last time I discussed this issue with @jdegoes, he said that it is better to maintain one version on Cats Effect 3.

**Comment by jdegoes** [comment]
I think we can choose the "compiling" one to be CE3, but still support CE2 migration guide that does _not_ compile.

**Review by jdegoes — APPROVED** [review]

**Pull request merged**